bob-apps/bin/bobd.rs

```diff
@@ -100,14 +108,78 @@ async fn main() {
 
     let factory = Factory::new(node.operation_timeout(), metrics);
     bob.run_periodic_tasks(factory);
-    let new_service = BobApiServer::new(bob);
-
-    Server::builder()
-        .tcp_nodelay(true)
-        .add_service(new_service)
-        .serve(addr)
-        .await
-        .unwrap();
+    let authentication_type = matches.value_of("authentication_type").unwrap();
+    match authentication_type {
+        "stub" => {
+            let bob_service = BobApiServer::new(bob);
```

Comment:
Both branches with different authenticators have plenty in common, I think it should be extracted into the function